From c5209dcad7874f9534f92978bdf606c73abf3f0e Mon Sep 17 00:00:00 2001
From: Gnoblin contributors <gnoblin@example.invalid>
Date: Sat, 12 Sep 2026 12:30:00 +0100
Subject: [PATCH] build: omit extension preferences service

GNOME 51 no longer builds the Extensions application, but Shell still ships
the D-Bus preference service's launcher, resource bundle, service activation
file, hidden desktop entry, and icons.

Remove those public entry points from a Gnoblin build as well.
---
 data/icons/meson.build      | 4 ++++
 data/meson.build            | 4 +---
 js/dbusServices/meson.build | 1 -
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/data/icons/meson.build b/data/icons/meson.build
index 277df01..b3929cb 100644
--- a/data/icons/meson.build
+++ b/data/icons/meson.build
@@ -1,4 +1,8 @@
 excluded_icons=[]
+excluded_icons += [
+  'scalable/apps/org.gnome.Shell.Extensions.svg',
+  'symbolic/apps/org.gnome.Shell.Extensions-symbolic.svg',
+]
 if not have_portal_helper
   excluded_icons += [
     'scalable/apps/org.gnome.Shell.CaptivePortal.svg',
diff --git a/data/meson.build b/data/meson.build
index 61bb876..ab21357 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,8 +1,6 @@
 data_builddir = meson.current_build_dir()
 
-desktop_files = [
-  'org.gnome.Shell.Extensions.desktop',
-]
+desktop_files = []
 service_files = []
 
 if have_portal_helper
diff --git a/js/dbusServices/meson.build b/js/dbusServices/meson.build
index 48b7f89..4bda8da 100644
--- a/js/dbusServices/meson.build
+++ b/js/dbusServices/meson.build
@@ -4,11 +4,10 @@ launcherconf.set('prefix', prefix)
 launcherconf.set('libdir', libdir)
 
 dbus_services = {
-  'org.gnome.Shell.Extensions': 'extensions',
   'org.gnome.Shell.Notifications': 'notifications',
   'org.gnome.Shell.Screencast': 'screencast',
   'org.gnome.ScreenSaver': 'screensaver',
 }
 
 config_dir = '@0@/..'.format(meson.current_build_dir())
 
-- 
2.55.0
